focus-android/app/src/main/java/org/mozilla/focus/activity/MainActivity.kt

```diff
@@ -71,6 +75,15 @@ open class MainActivity : LocaleAwareAppCompatActivity() {
     private lateinit var startupTypeTelemetry: StartupTypeTelemetry
     private var _binding: ActivityMainBinding? = null
     private val binding get() = _binding!!
+    private lateinit var privateNotificationObserver: PrivateNotificationFeature
+    private val notificationPermission =
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) { granted ->
+            when {
+                granted -> {
```

Comment:
For my understanding, we aren't handling `notGranted` and `shouldShowPermissionRationale()` cases, should we?

Reply:
If UX / Product decides we should notify the user of the consequences of not granting the notification permission, we could do that in the `notGranted` case. Currently, we have no other reason to handle that.